### PR TITLE
CA gateway

### PR DESCRIPTION
**Pull request by minijackson**
Package the CA PV gateway, with also a NixOS module for configuring it as a systemd service, and a setup guide in the documentation.

cc @agaget & @stephane-cea for review

### CODE REVIEW

File: nixos/modules/ca-gateway.nix [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      wantedBy = ["multi-user.target"];
+
+      # When initializing the IOC, PV Access looks for network interfaces that
+      # have IP addresses. "network.target" may be too early, especially for
+      # systems with DHCP.
+      wants = ["network-online.target"];
+      after = ["network-online.target"];
+
+      serviceConfig = {
+        ExecStart = "${pkg}/bin/gateway ${commandLine}";
```
Comment (minijackson) [inline nixos/modules/ca-gateway.nix:185]: Add the `Restart=always` option

**Review by agaget — COMMENTED** [review]
Comments

File: doc/nixos/guides/ca-gateway.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+you can examine the [gateway main page].
+
+  [gateway main page]: https://epics.anl.gov/extensions/gateway/
+
+# Pre-requisites
+
+-   Having a NixOS machine with a flake configuration.
+
+If you're not sure how to do this,
+you can follow the [Archiver Appliance tutorial],
```
[the remark this answers is not in the document]
Comment (minijackson) [reply doc/nixos/guides/ca-gateway.md:22]: TODO: NixOS + EPNix install guide

File: doc/nixos/guides/ca-gateway.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  };
+}
+```
+
+# Filtering IOCs
+
+By using the `cip` setting,
+you can filter which IOCs get exposed by the gateway.
+This is equivalent to setting the environment variable `EPICS_CA_ADDR_LIST`
+and setting `EPICS_CA_AUTO_LIST=NO`.
```
Comment (agaget) [inline doc/nixos/guides/ca-gateway.md:141]:
```suggestion
and setting `EPICS_CA_AUTO_ADDR_LIST=NO`.
```

**Review by agaget — APPROVED** [review]
stephane and I are happy about it ! 🥇

**Pull request merged**